pkgs/by-name/da/davis/package.nix

```diff
@@ -1,11 +1,13 @@
 {
   lib,
   fetchFromGitHub,
-  php,
+  php83,
   nixosTests,
 }:
 
-php.buildComposerProject2 (finalAttrs: {
+# NOTE: davis does not support php 8.4 yet
+#       https://github.com/tchapi/davis/issues/195
+php83.buildComposerProject2 (finalAttrs: {
```

Comment:
Referencing specific version of packages in implementation is not recommended. Favor referencing in the interface (or closer to the interface). See:
https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#changing-implicit-attribute-defaults

Reply:
`davis` isn't explicitly included in `all-packages.nix` because it is in the `by-name/`  tree. What's the right way to handle this?

Reply:
I think, that would be best to keep the package interface overridable. That's what I would do. By correct, I cannot say, because this is nixpkgs. Err...

Reply:
> `davis` isn't explicitly included in `all-packages.nix` because it is in the `by-name/` tree. What's the right way to handle this?

I cannot say what is correct for nixpkgs. I'd add to `all-packages.nix` since it is overriding packages. But hey. Don't quote me on this. Since I don't know what is 'right' here. Feel free to do nothing in case you find this too confusing. I pointed you to the docs I knew. Which seemed reasonable for me. Adding overrides to `all-packages.nix` is what I used to do. But I may be outdated by now since I'm out of the loop.

Reply:
> I'd add to all-packages.nix since it is overriding packages

This is what I've done! After reading the docs you linked again it very clearly says :

> Packages found in the name-based structure are automatically included, without needing to be added to all-packages.nix. However if the implicit attribute defaults need to be changed for a package, this must still be declared in all-packages.nix